src/vivarium_gates_mncnh/components/observers.py

```diff
@@ -80,3 +88,102 @@ def register_observations(self, builder: Builder) -> None:
+        }
+
+    def __init__(self):
+        super().__init__()
+        self.maternal_disorders = MATERNAL_DISORDERS
+
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+
+    def get_configuration(self, builder):
+        return builder.configuration["stratification"][self.get_configuration_name()]
+
+    def register_observations(self, builder: Builder) -> None:
+        pop_filter = f"{COLUMNS.ALIVE} == 'dead'"
+        builder.results.register_stratification(
+            "cause_of_maternal_death",
+            MATERNAL_DISORDERS + ["not_dead"],
+            excluded_categories=["not_dead"],
+            requires_columns=[COLUMNS.CAUSE_OF_DEATH],
+        )
+
+        builder.results.register_adding_observation(
+            name="maternal_disorder_deaths",
+            pop_filter=pop_filter,
+            requires_columns=[COLUMNS.ALIVE],
+            additional_stratifications=self.configuration.include
+            + ["cause_of_maternal_death"],
+            excluded_stratifications=self.configuration.exclude,
+            to_observe=self.to_observe,
+        )
+        builder.results.register_adding_observation(
+            name="maternal_disorder_ylls",
+            pop_filter=pop_filter,
+            requires_columns=[COLUMNS.ALIVE, COLUMNS.YEARS_OF_LIFE_LOST],
+            additional_stratifications=self.configuration.include
+            + ["cause_of_maternal_death"],
+            excluded_stratifications=self.configuration.exclude,
+            to_observe=self.to_observe,
+            aggregator=self.calculate_ylls,
+        )
+        for cause in self.maternal_disorders:
+            builder.results.register_adding_observation(
+                name=f"{cause}_counts",
+                pop_filter=f"{cause} == True",
+                requires_columns=[cause],
+                additional_stratifications=self.configuration.include,
+                excluded_stratifications=self.configuration.exclude,
+                to_observe=self.to_observe,
+            )
+            builder.results.register_adding_observation(
+                name=f"{cause}_ylds",
+                pop_filter=f'{COLUMNS.ALIVE} == "alive" and {cause} == True',
+                requires_columns=[cause],
+                additional_stratifications=self.configuration.include,
+                excluded_stratifications=self.configuration.exclude,
+                to_observe=self.to_observe,
+                aggregator=partial(self.calculate_ylds, cause=cause),
+            )
+
+    def to_observe(self, event: Event) -> bool:
+        return self._sim_step_name() == SIMULATION_EVENT_NAMES.MORTALITY
+
+    def calculate_ylls(self, data: pd.DataFrame) -> float:
+        return data[COLUMNS.YEARS_OF_LIFE_LOST].sum()
+
+    def calculate_ylds(self, data: pd.DataFrame, cause: str) -> float:
+        yld_per_case = self.lookup_tables[f"{cause}_ylds"](data.index)
+        return yld_per_case.sum()
```

Comment:
@rmudambi I think this is correct but let me reason it out here to see if you agree. Data is every simulant that will have a given maternal disorder so this will get the yld_per_case for each of these simulants and we will sum them. This would be the same as getting the total number of cases for each age group and multiplying it by the yld_per_case.

Reply:
Looks right to me